drivers/hwinfo/hwinfo_mcux_syscon.c

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2021 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/hwinfo.h>
+#include <string.h>
+#include <sys/byteorder.h>
+
+ssize_t z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length)
+{
+	uint32_t id[4];
+	uint32_t *idp = id;
+
+	*idp++ = sys_cpu_to_be32(SYSCTL0->UUID[0]);
+	*idp++ = sys_cpu_to_be32(SYSCTL0->UUID[1]);
+	*idp++ = sys_cpu_to_be32(SYSCTL0->UUID[2]);
+	*idp++ = sys_cpu_to_be32(SYSCTL0->UUID[3]);
+
```

Comment:
Any reason for the pointer and not just doing:

```
id[0] = ...
id[1] = ...
id[2] = ...
id[3] = ...
```

Reply:
Thank you for pointing this out, I have changed it.